builtin/provisioners/file/resource_provisioner.go

```diff
@@ -79,17 +79,11 @@ func applyFn(ctx context.Context) error {
 }
 
 func validateFn(d *schema.ResourceData) (ws []string, es []error) {
-	numSrc := 0
-	if _, ok := d.GetOk("source"); ok == true {
-		numSrc++
+	if d.Get("source").(string) == "" && d.Get("content").(string) == "" {
```

Comment:
What about tests? This functionality was broken for 2 releases since 0.9.6, and this is core functionality. Would be nice to catch this, so far I can see current tests are not catching the problem described in #15177